cdap-gateway/src/main/java/co/cask/cdap/gateway/discovery/UserServiceEndpointStrategy.java

```diff
@@ -89,4 +118,18 @@ public Discoverable pick() {
     }
     return result;
   }
+
+  private double fetchWeight(boolean routeConfigUnavailable, String version, RouteConfig routeConfig) {
+    if (routeConfigUnavailable || Strings.isNullOrEmpty(version)) {
```

Comment:
To be consistent with the rest of the logic, should just check if `version == null`.